mantidproject/mantid
# Indirect - ConvFit - Improve the PDF's generated using FABADA minimiser

**Description of work.**
This PR improves the output for PDF's (probability density functions) when using the FABADA minimiser in ConvFit.
Previously, the PDF's for each workspace would overwrite each other leading to just one output workspace in total, even though you may have 50 workspaces. The PDF workspaces also had the wrong names, and didn't have each row labelled - making it difficult to understand which row corresponds to which parameter.
There is also now a tick box to decide whether or not you want the PDF's to be calculated.

**To test:**
1. `Interfaces`->`Indirect`->`Data Analysis`->`ConvFit`
2. Load the data below
3. Fit Type: `One Lorentzian`
4. Delta Function: Yes
5. Flat Background
6. Minimizer: `FABADA`
7. PDF property should be true
8. Set `Fit String` to be `0,1`
9. Click `Run` and wait a while

- A group workspace with suffix `_PDFs` should be generated. Inside this group there should be two appropriately named workspaces. 
- Open one of these workspaces and make sure it has row headings. For example these headings should include a `f0.A0` heading and a `Chi Squared` heading at the bottom of the workspace. 
- Plot one of these spectra by right clicking on a row heading and select `Plot spectrum`. Find the most probably value on the x axis (where the peak is), and this should roughly correspond for the value of this parameter in the FitPropertyBrowser.
- Try running the fit again, but untick `PDF` so that the PDF's are not calculated.

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2477800/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2477799/irs26173_graphite002_res.zip)

Fixes #23759 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
